cmake/QSharp.cmake

```diff
@@ -0,0 +1,12 @@
+option (BUILD_SHARED_LIBS "Build shared libraries" ON)
+option (BUILD_DLL "Build DLL for Q# support" ON)
```

Comment:
BUILD_DLL or BUILD_QSHARP?

Reply:
The DLL theoretically has Windows OS application outside of QSharp. If this is a sticking point, I'll change it, but I slightly prefer it as-is.